lib/krane/kubernetes_resource.rb

```diff
@@ -35,6 +35,8 @@ class KubernetesResource
       If you have reason to believe it will succeed, retry the deploy to continue to monitor the rollout.
       MSG
 
+    ALLOWED_DEPLOY_METHOD_OVERRIDES = %w(create replace replace-force)
```

Comment:
I needed to get rid of `apply` here because
- It's the default method, anyway
- https://github.com/Shopify/krane/blob/master/lib/krane/resource_deployer.rb#L98: If you try to override the deploy method of a prunable type, it will still get applied, regardless of whichever override method you choose.

Note that the latter point means overriding will prevent default pruning behaviour __for the resource(s) whose deploy methods are being overridden__. This isn't bad, in and of itself, but it's worth pointing out.

Reply:
> Note that the latter point means overriding will prevent default pruning behaviour for the resource(s) whose deploy methods are being overridden. This isn't bad, in and of itself, but it's worth pointing out.

Worth making a note about this in a readme somewhere or nah?